streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -190,11 +190,42 @@ public void print(Serde<K> keySerde, Serde<V> valSerde) {
 
     @Override
     public void print(Serde<K> keySerde, Serde<V> valSerde, String streamName) {
+        print(keySerde, valSerde, streamName, null);
+    }
+
+    @Override
+    public void print(final Serde<K> keySerde, final Serde<V> valSerde, String streamName, final KeyValueMapper<K, V, String> mapper){
```

Comment:
Yes you're right. I'll propose this in a PR that concerns only the `peek` related change (KAFKA-4772).

_Forgive me for the combined PR – there were some uncertainties regarding both issues, KAFKA-4772 and KAFKA-4830, such that I thought a single [WIP] is the most convenient to discuss this._